docs/userGuide/syntax/questions.mbdf

```diff
@@ -1,130 +1,343 @@
+  </template>
 </question>
 </variable>
 </include>
+</panel>
+
+<box type="tip" seamless class="mt-3">
 
-**Use the minimized Panel component to create and group questions inline.**
+Placing the question into the header is entirely optional. You may also wish to include the question directly in the question body, omitting the header entirely.
+</box>
+
+
+****Common `<question>` Options and Slots****
+Name | Type | Default | Description
+--- | --- | --- | ---
+header | `String` | `''` | The markup to insert into the question header. The header is omitted if this is not provided.
+header | Slot | empty | Functions similar to the `header` option, and overrides it if both are present.
+hint  | `String` | `''` | The content to display in the hint box.
+hint  | Slot | empty | Functions similar to the `hint` option, and overrides it if both are present.
+
+
+#### Mcq and Checkbox Questions {.mt-4}
+
+Mcq and checkbox questions are indicated with the `type="mcq"` or `type="checkbox"` attribute.
+
+In both instances, you can include the possible answers using the `<q-option>` component, placed anywhere inside the <tooltip content="if you wish, you could place it in the `header` mentioned above as well!">question</tooltip>. To indicate the correct option(s), add the `<q-option correct>` attribute.
+
+Optionally, you can provide the reason for the particular option using the `<q-option reason="...">` attribute, or the `<template slot="reason">` slot for more expressive formatting, similar to the `hint` and `header` options and slots.
+
+**Mcq Question**
+
+<variable name="mcqQuestion"><question type="mcq" header="Which of these **contradicts** the heuristics recommended when creating test cases with multiple inputs?">
+  <!-- Insert the reason for the option using the reason attribute -->
+  <q-option reason="This is **correct**. We need to figure out if a positive test case works!">
+    Each valid test input should appear at least once in a test case that doesn’t have any invalid inputs.
+  </q-option>
+  <q-option>
+    It is ok to combine valid values for different inputs.
+  </q-option>
+  <q-option>
+    No more than one invalid test input should be in a given test case.
+  </q-option>
+  <!-- Use the 'correct' attribute to indicate an option as correct. -->
+  <q-option correct>
+    All invalid test inputs must be tested together.
+    <!-- Optionally, you may use a reason slot instead of a reason attribute. -->
+    <template slot="reason">
+    If you test all invalid test inputs together, you will not know if each one of the invalid inputs are handled
+    correctly by the SUT.
+    This is because most SUTs return an error message upon encountering the first invalid input.
+    </template>
+  </q-option>
+  <template slot="hint">
+  How do you figure out which inputs are wrong? (or correct)
+  </template>
+</question></variable>
```

Comment:
We don't use `template` anywhere else. Why not `<div slot="...">`?

Reply:
updated, note that it introduces an additional wrapper `<div>` though.

perhaps we could build a shorthand syntax feature for `<template slot="xx">` -> `<t slot="xx>`? (noting that vue 3.0 already shortens `slot="xx"` to `#xx`